javaslang/src/test/java/javaslang/collection/euler/Euler17Test.java

```diff
@@ -30,61 +31,124 @@
      */
     @Test
     public void shouldSolveProblem17() {
-        assertThat(letterCountInNumber(1)).isEqualTo(3);
-        assertThat(letterCountInNumber(342)).isEqualTo(23);
-        assertThat(letterCountInNumber(115)).isEqualTo(20);
-        assertThat(letterCountInNumbersUpToAndIncluding(5)).isEqualTo(19);
-        assertThat(letterCountInNumbersUpToAndIncluding(1000)).isEqualTo(21124);
+        runTestsFor(new SolutionA());
+        runTestsFor(new SolutionB());
+
+        // Additional capability, only for more general solution B
+        assertThat(new SolutionB().letterCount(Integer.MAX_VALUE)).isEqualTo("twobilliononehundredandfortysevenmillionfourhundredandeightythreethousandsixhundredandfortyseven".length());
     }
 
-    private static final int THOUSAND = "thousand".length();
-    private static final int HUNDRED = "hundred".length();
-    private static final int AND = "and".length();
-    private static final int[] LETTER_COUNT = new int[91];
-
-    static {
-        LETTER_COUNT[1] = "one".length();
-        LETTER_COUNT[2] = "two".length();
-        LETTER_COUNT[3] = "three".length();
-        LETTER_COUNT[4] = "four".length();
-        LETTER_COUNT[5] = "five".length();
-        LETTER_COUNT[6] = "six".length();
-        LETTER_COUNT[7] = "seven".length();
-        LETTER_COUNT[8] = "eight".length();
-        LETTER_COUNT[9] = "nine".length();
-        LETTER_COUNT[10] = "ten".length();
-        LETTER_COUNT[11] = "eleven".length();
-        LETTER_COUNT[12] = "twelve".length();
-        LETTER_COUNT[13] = "thirteen".length();
-        LETTER_COUNT[14] = "fourteen".length();
-        LETTER_COUNT[15] = "fifteen".length();
-        LETTER_COUNT[16] = "sixteen".length();
-        LETTER_COUNT[17] = "seventeen".length();
-        LETTER_COUNT[18] = "eighteen".length();
-        LETTER_COUNT[19] = "nineteen".length();
-        LETTER_COUNT[20] = "twenty".length();
-        LETTER_COUNT[30] = "thirty".length();
-        LETTER_COUNT[40] = "forty".length();
-        LETTER_COUNT[50] = "fifty".length();
-        LETTER_COUNT[60] = "sixty".length();
-        LETTER_COUNT[70] = "seventy".length();
-        LETTER_COUNT[80] = "eighty".length();
-        LETTER_COUNT[90] = "ninety".length();
+    private static void runTestsFor(SolutionProblem17 solution) {
+        List.of("one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten", "eleven", "twelve", "thirteen", "fourteen", "fifteen", "sixteen", "seventeen", "eighteen", "nineteen", "twenty", "twentyone")
+            .zipWithIndex().forEach(t -> {
+            final int number = t._2.intValue() + 1;
+            final String numberAsString = t._1;
+            assertThat(numberAsString).hasSize(solution.letterCount(number));
+        });
```

Comment:
This is how Idea formats it, if I put the forEach on that line.
Will put in on the next :)